FEXCore/Source/Interface/Core/LookupCache.h

```diff
@@ -15,24 +17,48 @@
 #include <mutex>
 
 namespace FEXCore {
-
 struct LookupCacheWriteLockToken {
+  // Move-only type
+  LookupCacheWriteLockToken(const LookupCacheWriteLockToken&) = delete;
+  LookupCacheWriteLockToken& operator=(const LookupCacheWriteLockToken&) = delete;
+  LookupCacheWriteLockToken(LookupCacheWriteLockToken&& rhs) = delete;
+  LookupCacheWriteLockToken& operator=(LookupCacheWriteLockToken&&) = delete;
+private:
+  // Only constructible by GuestToHostMap
+  friend struct GuestToHostMap;
+  LookupCacheWriteLockToken(FEXCore::Utils::WritePriorityMutex::Mutex& Mutex)
+    : Lock {Mutex} {}
+  std::lock_guard<FEXCore::Utils::WritePriorityMutex::Mutex> Lock;
+};
+
+struct LookupCacheReadLockToken {
+  // Move-only type
+  LookupCacheReadLockToken(const FEXCore::Utils::WritePriorityMutex::Mutex&) = delete;
+  LookupCacheReadLockToken& operator=(const FEXCore::Utils::WritePriorityMutex::Mutex&) = delete;
+  LookupCacheReadLockToken(FEXCore::Utils::WritePriorityMutex::Mutex&& rhs) = delete;
+  LookupCacheReadLockToken& operator=(FEXCore::Utils::WritePriorityMutex::Mutex&&) = delete;
+
 private:
   // Only constructible by GuestToHostMap
   friend struct GuestToHostMap;
-  LookupCacheWriteLockToken(std::mutex& Mutex)
+  LookupCacheReadLockToken(FEXCore::Utils::WritePriorityMutex::Mutex& Mutex)
     : Lock {Mutex} {}
-  std::lock_guard<std::mutex> Lock;
+  std::shared_lock<FEXCore::Utils::WritePriorityMutex::Mutex> Lock;
 };
 
 struct GuestToHostMap {
-  std::mutex WriteLock;
+  FEXCore::Utils::WritePriorityMutex::Mutex WriteLock {};
```

Comment:
Calling it WriteLock when it's actually used to guard reads too is misleading:
```suggestion
  FEXCore::Utils::WritePriorityMutex::Mutex Lock {};
```